### ISSUE TITLE
Installation of root certificate on an Android VM doesn't work

### ISSUE THREAD

**Issue by pyretta** [MegatronKing/HttpCanary#102]
Hi,
I can't install the root certificate on an Android VM. Everytime I try, I am stuck in an infinite loop of PIN requests to unlock the screen. Tried it on a VirtualBox Android 7.1 x86 (32bit) System which is rooted. I checked the root status with "Root Checker Basic".

On a physical device (Samsung Galaxy S7 - Android 8.1 - without root) I can install the CA and can monitor the traffic of apps without problems. 

In your description you said it would be compatible to "any android arm or x86 devices include emulators". So this should be working with my VM, right? 

If so, what am I missing? 
What should I configure in the VM settings or App settings?

Thank you very much for any help and advice.

Btw.: Very nice app! If I can use it with my VM I will buy the PRO version, because it seems to be really good. As I understood, in the PRO Version I could export the data streams into excel, if I install the plugin, right? This would be great and could be used for our app testings. Thanks again for your advice.

**Comment by MegatronKing** [comment]
"an infinite loop of PIN requests to unlock the screen" that seems to be a system bug. I think you can preset the PIN in system Settings before running the app.

**Comment by pyretta** [comment]
Thank you for your quick reply. What exactly do you mean with "system bug" and how could I preset the PIN in settings? Note that the described behave only occured if I am trying to install your root CA. Everything else runs smoothly - other apps, google play store etc. Thank you for your advice.

**Comment by MegatronKing** [comment]
Open Settings -> Security -> Screen Lock -> PIN, set the PIN. And then open HttpCanary and install certificate.

**Comment by pyretta** [comment]
Thanks a lot. That's exactly the way I've been trying to do it. It's the way how I set the PIN. In "Security > Credentials Store > Trusted Credentials" (translated from German) your certificate already appears. But in your app I still can't install the root certificate.

**Comment by MegatronKing** [comment]
Apart from PIN, did you try Pattern or Password as the screen lock?

**Comment by pyretta** [comment]
Ok. I tried the other options and with "Password" instead of PIN I have to Enter the Password twice, then it appears the Certificate install popup, I accept everything and let it as it is, press ok. Then I'm again in a infinite loop to enter the password until I close your App. If I open your App again it says no root certificate installed.

**Comment by MegatronKing** [comment]
I can't figure out why, maybe you can increase the memory quota of the simulator.

**Comment by MegatronKing** [comment]
If you are an android developer, see this code: https://github.com/MegatronKing/NetBare/blob/master/netbare-core/src/main/java/com/github/megatronking/netbare/ssl/CertificateInstallActivity.java#L52

**Comment by pyretta** [comment]
the specs are: 4 GB ram, 2 CPUs, 500 GB hdd. Do I need more ram or CPUs? It's been a long time since I have programmed Java or android natively, so unfortunately I can do little with the code.

**Comment by pyretta** [comment]
Would it be helpful if I make some screenshots? But the most of it would be in German language i suppose ...

**Comment by pyretta** [comment]
Here are some screenshots with system settings, app settings, and the certificate install prozess:
![HTTPcanary_supportrequest_img01](https://user-images.githubusercontent.com/11720167/78158225-45be0a00-7441-11ea-8537-a72242007094.png)
![HTTPcanary_supportrequest_img02](https://user-images.githubusercontent.com/11720167/78158230-4656a080-7441-11ea-81c5-0864d73d0330.png)
![HTTPcanary_supportrequest_img03](https://user-images.githubusercontent.com/11720167/78158232-46ef3700-7441-11ea-87de-4e7191229a1f.png)
![HTTPcanary_supportrequest_img04](https://user-images.githubusercontent.com/11720167/78158234-46ef3700-7441-11ea-96b9-fccd2811fd83.png)
![HTTPcanary_supportrequest_img05](https://user-images.githubusercontent.com/11720167/78158236-4787cd80-7441-11ea-8f7d-b00777a46c55.png)
![HTTPcanary_supportrequest_img06](https://user-images.githubusercontent.com/11720167/78158237-4787cd80-7441-11ea-81f9-e0dcfa3ed962.png)
![HTTPcanary_supportrequest_img07](https://user-images.githubusercontent.com/11720167/78158238-4787cd80-7441-11ea-99e4-e30f8a322055.png)
![HTTPcanary_supportrequest_img08](https://user-images.githubusercontent.com/11720167/78158239-48206400-7441-11ea-8bd9-60b01af07f66.png)
![HTTPcanary_supportrequest_img09](https://user-images.githubusercontent.com/11720167/78158243-48206400-7441-11ea-8580-91a3603b44f4.png)
![HTTPcanary_supportrequest_img10](https://user-images.githubusercontent.com/11720167/78158246-48206400-7441-11ea-8847-f0ee441bd959.png)
![HTTPcanary_supportrequest_img11](https://user-images.githubusercontent.com/11720167/78158249-48b8fa80-7441-11ea-8104-0b0afd59107f.png)
![HTTPcanary_supportrequest_img12](https://user-images.githubusercontent.com/11720167/78158253-48b8fa80-7441-11ea-9265-7d8c6e3cb3b6.png)

**Comment by MegatronKing** [comment]
Thanks for the feedback. I think this has been fixed on v3.3.1.

**Comment by pyretta** [comment]
Sorry to say this, but the issue isn't fixed yet. I have installed your mentioned Version 3.3.1, but I still can't install the root CA and therefore I still can't use your App. I waited with my answer because I hoped that it was a bug and with a further update it were fixed. But nope. It's still a problem. And as I have found here, I'm not the only one who has this problem: https://github.com/MegatronKing/HttpCanary/issues/107#issue-600445772
